qiskit/algorithms/evolvers/variational/__init__.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Variational Quantum Time Evolutions (:mod:`qiskit.algorithms.evolvers.variational`)
+===================================================================================
+
+Algorithms for performing Variational Quantum Time Evolution of quantum states,
+which can be tailored to near-term devices.
+:class:`~qiskit.algorithms.evolvers.variational.VarQTE` base class exposes an interface, compliant
+with the Quantum Time Evolution Framework in Qiskit Terra, that is implemented by
+:class:`~qiskit.algorithms.VarQRTE` and :class:`~qiskit.algorithms.VarQITE` classes for real and
+imaginary time evolution respectively. The variational approach is taken according to a variational
+principle chosen by a user.
+
+Examples:
+
+.. code-block::
+
+    from qiskit import BasicAer
+    from qiskit.circuit.library import EfficientSU2
+    from qiskit.opflow import SummedOp, I, Z, Y, X
+    from qiskit.algorithms.evolvers.variational.variational_principles import (
+        ImaginaryMcLachlanPrinciple,
+    )
+    from qiskit.algorithms import EvolutionProblem
+    from qiskit.algorithms import VarQITE
+
+    # define a Hamiltonian
+    observable = SummedOp(
+        [
+            0.2252 * (I ^ I),
+            0.5716 * (Z ^ Z),
+            0.3435 * (I ^ Z),
+            -0.4347 * (Z ^ I),
+            0.091 * (Y ^ Y),
+            0.091 * (X ^ X),
+        ]
+    ).reduce()
+
+    # define a parametrized initial state to be evolved
+    d = 1
+    ansatz = EfficientSU2(observable.num_qubits, reps=d)
```

Comment:
Not that it really is too significant but I am wondering why define d and then use it once for reps. Why not just do reps=1 directly?

For EvolutionProblem below, since the parameters are being passed positionally defining time=1 and passing it as time makes things clear (as otherwise it would be passing it by value as 1 which would not be helpful to readers). But in the case of reps the usage is clear to readers I think.

Reply:
True, I am changing it.